atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceIntegrationTest.java

```diff
@@ -358,6 +365,78 @@ public void upgradeFromOlderInternalSchemaDoesNotErrorOnTablesWithOldMetadata()
         verify(logger, never()).error(anyString(), any(Object.class));
     }
 
+    @Test
```

Comment:
I think we should consolidate the tests here and just check the unwrapping logic in the the other test class